sdb/command.py

```diff
@@ -634,24 +634,31 @@ def pretty_print(self, objs: Iterable[drgn.Object]) -> None:
         # pylint: disable=missing-docstring
         raise NotImplementedError
 
-    def _call(  # type: ignore[return]
-            self,
-            objs: Iterable[drgn.Object]) -> Optional[Iterable[drgn.Object]]:
+    def check_input_type(self,
+                         objs: Iterable[drgn.Object]) -> Iterable[drgn.Object]:
         """
-        This function will call pretty_print() on each input object,
-        verifying the types as we go.
+        This function acts as a generator, checking that each passed object
+        matches the input type for the command
         """
-
         assert self.input_type is not None
         type_name = type_canonicalize_name(self.input_type)
         for obj in objs:
             if type_canonical_name(obj.type_) != type_name:
                 raise CommandError(
                     self.name,
-                    f'exepected input of type {self.input_type}, but received '
+                    f'expected input of type {self.input_type}, but received '
                     f'type {obj.type_}')
+            yield obj
 
-            self.pretty_print([obj])
+    def _call(  # type: ignore[return]
```

Comment:
Could you explain a bit more about ignoring the type error here? Does `mypy` raise errors if you were to do `-> None` ?

Reply:
I didn't add that annotation - it was in the existing code. I can experiment with passing `None` though.

Reply:
It seems we do this to allow commands to be both a `PrettyPrinter` and `Locator`.

Changing the return type to `None`, we see the error: `definition of "_call" in base class "Locator" is incompatible with definition in base class "PrettyPrinter"`

Reply:
Thanks for looking into this! You can go ahead and merge. Since this is something that was already in the existing code it can be tackled under https://github.com/delphix/sdb/issues/195.